inc/saigenericprogrammable.h

```diff
@@ -0,0 +1,155 @@
+/**
+ * Copyright (c) 2014 Microsoft Open Technologies, Inc.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License"); you may
+ *    not use this file except in compliance with the License. You may obtain
+ *    a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    THIS CODE IS PROVIDED ON AN *AS IS* BASIS, WITHOUT WARRANTIES OR
+ *    CONDITIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT
+ *    LIMITATION ANY IMPLIED WARRANTIES OR CONDITIONS OF TITLE, FITNESS
+ *    FOR A PARTICULAR PURPOSE, MERCHANTABILITY OR NON-INFRINGEMENT.
+ *
+ *    See the Apache Version 2.0 License for specific language governing
+ *    permissions and limitations under the License.
+ *
+ *    Microsoft would like to thank the following companies for their review and
+ *    assistance with these files: Intel Corporation, Mellanox Technologies Ltd,
+ *    Dell Products, L.P., Facebook, Inc., Marvell International Ltd.
+ *
+ * @file    saigenericprogrammable.h
+ *
+ * @brief   This module defines SAI Genetic Programmable Extensions (GPE)
+ */
+
+#if !defined (__SAIGENERICPROGRAMMABLE_H_)
+#define __SAIGENERICPROGRAMMABLE_H_
+
+#include <saitypes.h>
+
+/**
+ * @defgroup SAIGENERICPROGRAMMABLE SAI - GPE specific API definitions
+ *
+ * @{
+ */
+
+/**
+ * @brief Attribute Id for Generic Programmable extension
+ */
+typedef enum _sai_generic_programmable_attr_t
+{
+    /**
+     * @brief Start of attributes
+     */
+    SAI_GENERIC_PROGRAMMABLE_ATTR_START,
+
+    /**
+     * @brief HW block name to program the entry
+     *
+     * @type sai_s8_list_t
+     * @flags MANDATORY_ON_CREATE | CREATE_ONLY
+     */
+    SAI_GENERIC_PROGRAMMABLE_ATTR_OBJECT_NAME = SAI_GENERIC_PROGRAMMABLE_ATTR_START,
+
+    /**
+     * @brief JSON string carrying HW block entry information
+     *
+     * @type sai_json_t
+     * @flags CREATE_AND_SET
+     * @default vendor
+     */
+    SAI_GENERIC_PROGRAMMABLE_ATTR_ENTRY,
```

Comment:
If not already done, please add API examples that cover:

1) creating multiple entries of the same object name. For example, creating multiple VIP entries.


2) Updating a single attribute of an entry. For example, changing the nexthop-id of a VIP entry.

Reply:
Hi, I don't see example of "creating multiple VIP entries" in https://github.com/opencomputeproject/SAI/blob/19ba5116264829d3a7b933212ab6bd0d977a3a55/doc/SAI_Proposal_Generic_Extensions.md, and also changing the entry?

Reply:
Adding multiple different entries is the same as the single create example. The same invocation as existing example would work with modified IP and nexthop information.
I added an code example for update scenario also.